### PR TITLE
Added additional Bonsai docs to default_prover 

### PR DESCRIPTION
**Pull request by mothran**
This PR expands on the docs in places where we depend on external inputs like BONSAI_API_URL and BONSAI_API_KEY env vars to make it more clear their relationship to the selected prover.

### CODE REVIEW

**Review by flaub — APPROVED** [review]

File: risc0/zkvm/src/host/client/prove/bonsai.rs [hunk 1]
```diff
@@ -22,6 +22,9 @@ use super::Prover;
 use crate::{ExecutorEnv, ProverOpts, Receipt, VerifierContext};
 
 /// An implementation of a [Prover] that runs proof workloads via Bonsai.
+///
+/// Requires a BONSAI_API_URL and BONSAI_API_KEY environment variables to
+/// submit proving sessions to Bonsai.
```
Comment (tzerrell) [inline risc0/zkvm/src/host/client/prove/bonsai.rs:27]:
```suggestion
/// Requires BONSAI_API_URL and BONSAI_API_KEY environment variables to
/// submit proving sessions to Bonsai.
```

**Review by tzerrell — APPROVED** [review]
Thank you!

File: risc0/zkvm/src/host/client/prove/bonsai.rs (hunk 1 above)
Comment (flaub) [reply risc0/zkvm/src/host/client/prove/bonsai.rs:27]: Also would be good to wrap env var names with backticks for formatting.

Comment (tzerrell) [reply risc0/zkvm/src/host/client/prove/bonsai.rs:27]: Good point
```suggestion
/// Requires `BONSAI_API_URL` and `BONSAI_API_KEY` environment variables to
/// submit proving sessions to Bonsai.
```

File: risc0/zkvm/src/host/client/prove/mod.rs [hunk 2]
```diff
@@ -127,6 +127,10 @@ impl Default for ProverOpts {
 
 /// Return a default [Prover] based on environment variables, falling back to a
 /// default CPU-based prover.
+///
+/// While not in RISC0_DEV_MODE, if the `BONSAI_API_URL` and `BONSAI_API_KEY` environment variables
```
Comment (tzerrell) [inline risc0/zkvm/src/host/client/prove/mod.rs:131]:
```suggestion
/// While not in `RISC0_DEV_MODE`, if the `BONSAI_API_URL` and `BONSAI_API_KEY` environment variables
```

**Pull request merged**